pages/preso.tsx

```diff
@@ -5,34 +5,40 @@ import { PresoForm } from '@types'
 import { NextPage } from 'next'
 import Head from 'next/head'
 import { useRouter } from 'next/router'
+import {validatePresos} from "validation/presos"
 
 const Preso: NextPage = () => {
   const router = useRouter()
   const { session } = useSupabase()
 
   const onSubmit = async (values: PresoForm) => {
+    const { title, url, eventName, eventLocation } = values;
     try {
-      const response = await fetch('/api/preso', {
-        method: 'POST',
-        body: JSON.stringify({
-          ...values,
-          userId: session!.user?.id
-        } as PresoForm)
-      })
-      const json = await response.json()
-      router.replace(`/qr?preso=${encodeURIComponent(json.presoShortCode)}`)
+      const isPresosvalid = await validatePresos.validate({ title, url, eventName, eventLocation });
+      if (isPresosvalid) {
+        const response = await fetch('/api/preso', {
+          method: 'POST',
+          body: JSON.stringify({
+            ...values,
+            userId: session!.user?.id
+          } as PresoForm)
+        })
+        const json = await response.json()
+        router.replace(`/qr?preso=${encodeURIComponent(json.presoShortCode)}`)
+      }
     } catch (error) {
       const { message } = error as Error
+      alert(message)
       console.error(message)
     }
   }
 
   return (
     <>
-      <h1 className="text-3xl bg-black py-2 px-6 text-white">
+      <h1 className="text-2xl bg-black py-2 px-6 text-white text-center font-bold">
```

Comment:
The changes are unrelated to the validation and should happen in a separate PR in the future.